flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java

```diff
@@ -18,176 +18,905 @@
+                        .testCase(VARCHAR(5), "Flink", new byte[] {70, 108, 105, 110, 107})
+                        // https://issues.apache.org/jira/browse/FLINK-24419 - not trimmed to 2
+                        // bytes
+                        .testCase(STRING(), "Apache", new byte[] {65, 112, 97, 99, 104, 101})
+                        // Not supported
+                        // BOOLEAN
+                        .testCase(BINARY(2), new byte[] {0, 1}, new byte[] {0, 1})
+                        .testCase(VARBINARY(3), new byte[] {0, 1, 2}, new byte[] {0, 1, 2})
+                        .testCase(BYTES(), new byte[] {0, 1, 2, 3, 4}, new byte[] {0, 1, 2, 3, 4})
+                        // Not supported - no fix
+                        // DECIMAL
+                        // TINYINT
+                        // SMALLINT
+                        // INT
+                        // BIGINT
+                        // FLOAT
+                        // DOUBLE
+                        // DATE
+                        // TIME
+                        // TIMESTAMP
+                        // TIMESTAMP_WITH_TIME_ZONE
+                        // TIMESTAMP_LTZ
+                        // INTERVAL(YEAR())
+                        // INTERVAL(DAY())
+                        // ARRAY
+                        // MULTISET
+                        // MAP
+                        // ROW
+                        //
+                        // https://issues.apache.org/jira/browse/FLINK-24418
+                        // .testCase(RAW(byte[].class), new byte[] {1}, new byte[] {1})
+                        .build(),
+                CastTestSpecBuilder.test(BYTES(), "To BYTES")
+                        .testCase(CHAR(3), "foo", new byte[] {102, 111, 111})
+                        .testCase(VARCHAR(5), "Flink", new byte[] {70, 108, 105, 110, 107})
+                        .testCase(STRING(), "Apache", new byte[] {65, 112, 97, 99, 104, 101})
+                        // Not supported - no fix
+                        // BOOLEAN
+                        .testCase(BINARY(2), new byte[] {0, 1}, new byte[] {0, 1})
+                        .testCase(VARBINARY(3), new byte[] {0, 1, 2}, new byte[] {0, 1, 2})
+                        .testCase(BYTES(), new byte[] {0, 1, 2, 3, 4}, new byte[] {0, 1, 2, 3, 4})
+                        // Not supported - no fix
+                        // DECIMAL
+                        // TINYINT
+                        // SMALLINT
+                        // INT
+                        // BIGINT
+                        // FLOAT
+                        // DOUBLE
+                        // DATE
+                        // TIME
+                        // TIMESTAMP
+                        // TIMESTAMP_WITH_TIME_ZONE
+                        // TIMESTAMP_LTZ
+                        // INTERVAL(YEAR())
+                        // INTERVAL(DAY())
+                        // ARRAY
```

Comment:
let's support casting to `ARRAY<NUMERIC>` or at least `ARRAY<TINYINT>`? otherwise a user cannot use all the built-in array function to access elements or cardinality etc.

Reply:
ignore this comment, I forgot that we are casting to BYTES not from BYTES